src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,128 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState } from 'react';
+import { FilterType } from '../../types/FilterType';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  setVisibleTodos: (todos: Todo[]) => void,
+  visibleTodos: Todo[]
+};
+
+export const TodoFilter: React.FC<Props> = ({ todos, setVisibleTodos }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [isActive, setIsActive] = useState(false);
+  const [isCompleted, setIsCompleted] = useState(false);
+  let filteredTodos = [];
```

Comment:
Don't declare variables inside react component like this. They'll be redeclared on every render.

Reply:
@qoa11a could you explain how to do it, please

Reply:
`let filteredTodos = [];` - move to 10 line.